src/NightwatchServiceProvider.php

```diff
@@ -499,7 +499,11 @@ private function executionState(): RequestState|CommandState
                 currentExecutionStageStartedAtMicrotime: $this->timestamp,
                 deploy: $this->nightwatchConfig['deployment'] ?? '',
                 server: $this->nightwatchConfig['server'] ?? '',
-                user: new UserProvider($auth, fn () => $this->core->userDetailsResolver, fn () => $this->core->report(...)),
+                user: new UserProvider(
+                    fn (callable $callback) => $this->core->ignore(static fn () => $callback($auth)),
+                    fn () => $this->core->userDetailsResolver,
+                    fn () => $this->core->report(...),
+                ),
```

Comment:
This new callable is a helper that ensures all interactions with the auth manager are ignored.

I like this approach because it means we don't introduce a non-runtime circular dependency. We also put it in a closure because `$this->core` is not yet set.